topics/statistics/tutorials/intro-to-ml-with-python/tutorial.md

```diff
@@ -64,543 +92,483 @@ plt.rc('ytick.major', size=8, pad=12)
+### Approach 2: adding regularization and validation set
 
- * https://www.datacamp.com/community/tutorials/tutorial-ridge-lasso-elastic-net
+In the case of a Least Square fit, the function you are minimizing looks like:
 
- * https://towardsdatascience.com/regularization-in-machine-learning-76441ddcf99a
+$$ \sum_i (y_i-f(\pmb X_i,\pmb{\beta}))^2 $$
 
+, so the sum of squared difference between the observation and the predictions of your model.
 
-In case of a logistic regression you want to maximize your log likelihood which is now penalized by one of those functions:
 
-$\sum_{i}log(p_{i}) - \alpha\sum_{j}|\beta_{j}|$ , **l1 regularization** (Lasso) 
+**Regularization** is a way to reduce overfitting, and in the case of the linear model
+we do so by adding to this function a **penalization term which depends on coefficient weights**.
 
-$\sum_{i}log(p_{i}) - \frac{\alpha}{2}\sum_{j}\beta_{j}^{2}$ , **l2 regularization** (Ridge) 
+In brief, the stronger the coefficient, the higher the penalization. So only coefficients which bring more fit than penalization will be kept.
 
-$\sum_{i}log(p_{i}) - \alpha\sum_{j}(\rho|\beta_{j}|+\frac{(1-\rho)}{2}\beta_{j}^{2})$ , **elasticnet**
+> <comment-title></comment-title>
+> We report here the formulas used in `scikit-learn` functions. Other libraries may have a different parameterization, but the concepts stay the same.
+{: .comment}
 
-Rule is : **when you hypothesize that you have sparse features and so you believe that among all those features only a small subset is going to be interesting (but of course you don't know which ones...) then you try to use the regularization that will tend to put more of your features at the zero weight (the l1 regularization) and so reduce the complexity of your model.** This l1 norm that collapses non-important features to zero is another way to do feature selection.
 
 
-Now, we need a way to find this coefficient $\alpha$ which will set the strength of our regularization. This parameter is called an **hyperparameter**, and cannot be found directly like the others, since even if it is part of a new model it serve a generalization purpose and so should not be found by optimization on our full dataset. To do that on top of our first splitting between train dataset and test dataset, we will need to perfom some more splitting of our train data set.
+$$\frac{1}{2n}\sum_i (y_i-f(\pmb X_i,\pmb{\beta}))^2 + \alpha\sum_{j}\lvert\beta_{j}\rvert$$ , **l1 regularization** (Lasso) $$\alpha$$ being the weight that you put on that regularization 
 
-> <comment-title></comment-title>
->
-> the polynomial number we were using before is also an hyperparameter and can be find by the same technic consisting of splitting our data. Later on we will see other hyperparameters that are either related to model choice or regularization or intrically both.
->
-{: .comment}
+$$\sum_i (y_i-f(\pmb X_i,\pmb{\beta}))^2 + \alpha\sum_{j}\beta_{j}^{2}$$ , **l2 regularization** (Ridge) 
```

Comment:
ok, to fix your subscripts, please use delimiters `\\(` and `\\)` instead of the `$$`

so like this:

```suggestion
\\(\sum_i (y_i-f(\pmb X_i,\pmb{\beta}))^2 + \alpha\sum_{j}\beta_{j}^{2}\\) , **l2 regularization** (Ridge) 
```
This looks better now:

![image](https://github.com/user-attachments/assets/b294c8a5-4ee8-4d5c-a99c-66c3a8b4e444)

See also: https://training.galaxyproject.org/training-material/topics/contributing/tutorials/create-new-tutorial-content/tutorial.html#writing-mathematical-expressions